src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1523,14 +1525,21 @@ internal bool HasTopLevelNullabilityImplicitConversion(TypeWithAnnotations sourc
                 return true;
             }
 
-            if (source.IsPossiblyNullableTypeTypeParameter() && !destination.IsPossiblyNullableTypeTypeParameter())
+            if (IsPossiblyNullableTypeTypeParameter(source) && !IsPossiblyNullableTypeTypeParameter(destination))
             {
                 return false;
             }
 
             return !source.NullableAnnotation.IsAnnotated();
         }
 
+        private static bool IsPossiblyNullableTypeTypeParameter(in TypeWithAnnotations typeWithAnnotations)
+        {
+            var type = typeWithAnnotations.Type;
+            return type is object &&
```

Comment:
>return type is object && [](start = 12, length = 24)

It looks like we dropped `IsNotAnnotated` check. Is this intentional? Could you please elaborate?

Reply:
> It looks like we dropped IsNotAnnotated check. Is this intentional? 

Yes. Callers above are explicitly checking `NullableAnnotation`, and requiring `NullableAnnotation == NullableAnnotation.NotAnnotated` here would mean `T` and `T?` would be considered identical in `HasTopLevelNullabilityIdentityConversion()`. For instance, that would allow assigning `IEnumerable<T?>` to `IEnumerable<T>` in `UnconstrainedTypeParameter_21()`.

---
In reply to: [456696256](https://github.com/dotnet/roslyn/pull/45993#discussion_r456696256) [](ancestors = 456696256)